update rustc 1.20.0-nightly (d84693b93 2017-07-09)  to  complie error

error[E0425]: cannot find function `allocate` in module `heap`
   --> /home/room/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/capnp-nonblock-0.3.0/src/buf.rs:187:31
    |
187 |             let bytes = heap::allocate(len, refcount_len);
    |                               ^^^^^^^^ not found in `heap`

error[E0425]: cannot find function `oom` in module `alloc`
   --> /home/room/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/capnp-nonblock-0.3.0/src/buf.rs:188:50
    |
188 |             if bytes == ptr::null_mut() { alloc::oom() }
    |                                                  ^^^ not found in `alloc`

error[E0425]: cannot find function `deallocate` in module `heap`
   --> /home/room/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/capnp-nonblock-0.3.0/src/buf.rs:226:23
    |
226 |                 heap::deallocate(allocation, self.len + refcount_len, refcount_len);
    |                       ^^^^^^^^^^ not found in `heap`

error: use of unstable library feature 'allocator_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector (see issue #32838)
 --> /home/room/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/capnp-nonblock-0.3.0/src/buf.rs:1:19
  |
1 | use alloc::{self, heap};
  |                   ^^^^
  |
  = help: add #![feature(allocator_api)] to the crate attributes to enable

--- a/src/buf.rs
+++ b/src/buf.rs
@@ -1,4 +1,4 @@
-use alloc::{self, heap};
+use alloc::heap::{Heap, Alloc, Layout};
 use std::{cmp, io, mem, ops, ptr, slice};
 
 use std::io::Write;
@@ -184,8 +184,10 @@ impl RawBuf {
             // refcount, as well as required by Cap'n Proto. This requirement is
             // the primary reason that the raw allocation APIs are used instead
             // of something like RawVec.
-            let bytes = heap::allocate(len, refcount_len);
-            if bytes == ptr::null_mut() { alloc::oom() }
+            let bytes = match Heap.alloc(Layout::from_size_align(len, refcount_len).unwrap()) {
+                Result::Ok(val) => val,
+                Result::Err(err) => Heap.oom(err),
+            };
             *(bytes as *mut u64) = 1;
             RawBuf {
                 bytes: bytes.offset(refcount_len as isize),
@@ -223,7 +225,7 @@ impl Drop for RawBuf {
             let refcount = allocation as *mut u64;
             *refcount -= 1;
             if *refcount == 0 {
-                heap::deallocate(allocation, self.len + refcount_len, refcount_len);
+                Heap.dealloc(allocation,  Layout::from_size_align(self.len + refcount_len, refcount_len).unwrap());
             }
         }
     }

--- a/src/buf.rs
+++ b/src/buf.rs
@@ -74,7 +74,7 @@ impl MutBuf {
                                                            "failed to fill whole buffer")),
                     n => {
                         self.offset += n;
-                        let mut tmp = buf;
+                        let tmp = buf;
                         buf = &mut tmp[n..];
                     },
                 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -2,7 +2,7 @@
 //! writing [Cap'n Proto](https://capnproto.org/) messages to non-blocking
 //! streams.
 
-#![feature(alloc, heap_api, oom)]
+#![feature(alloc, allocator_api)]
 
 extern crate alloc;
 extern crate byteorder;

